src/main/java/org/testeditor/fixture/web/WebDriverFixture.java

```diff
@@ -532,149 +545,189 @@ public void closeBrowser() {
      * This method waits explicitly for a WebElement in the DOM-Object until the
      * given timeOut is reached before an Exception is thrown.
      * 
-     * @param elementLocator
-     *            Locator for Gui-Widget
-     * @param locatorType
-     *            Type of locator for Gui-Widget
-     * @param timeOutInSeconds
-     *            The max timeout in seconds when an element is expected before a
-     *            NotFoundException will be thrown.
+     * @param elementLocator Locator for Gui-Widget
+     * @param locatorType Type of locator for Gui-Widget
+     * @param timeOutInSeconds The max timeout in seconds when an element is
+     *            expected before a NotFoundException will be thrown.
      */
     @FixtureMethod
-    public void waitUntilElementFound(String elementLocator, LocatorStrategy locatorType, int timeOutInSeconds) {
+    public void waitUntilElementFound(String elementLocator, LocatorStrategy locatorType, int timeOutInSeconds)
+            throws FixtureException {
         WebDriverWait webDriverWait = new WebDriverWait(driver, timeOutInSeconds);
-        webDriverWait.until(ExpectedConditions.presenceOfElementLocated(getBy(elementLocator, locatorType)));
+        try {
+            webDriverWait.until(ExpectedConditions.presenceOfElementLocated(getBy(elementLocator, locatorType)));
+        } catch (TimeoutException e) {
+            throw new FixtureException("timeout during wait for element", //
+                    keyValues("elementLocator", elementLocator, //
+                            "locatorType", locatorType.toString(), //
+                            "timeout", Long.valueOf(timeOutInSeconds)),
+                    e);
+        }
     }
 
     /**
      * press enter on a specified Gui-Widget
      * 
-     * @param elementLocator
-     *            Locator for Gui-Widget
-     * @param locatorType
-     *            Type of locator for Gui-Widget
+     * @param elementLocator Locator for Gui-Widget
+     * @param locatorType Type of locator for Gui-Widget
      */
     @FixtureMethod
-    public void pressEnterOn(String elementLocator, LocatorStrategy locatorType) {
+    public void pressEnterOn(String elementLocator, LocatorStrategy locatorType) throws FixtureException {
         WebElement element = getWebElement(elementLocator, locatorType);
-        element.submit();
+        try {
+            element.submit();
+        } catch (NoSuchElementException e) {
+            throw new FixtureException("element seems not to be part of a form and cannot pre pressed enter on", //
```

Comment:
msg seems a bit weird, maybe change to something like:
_element seems not to be part of a form so enter cannot be pressed on it_